[LLVMLibUnwind] Add version 14.0.6

ararslan opened the pull request.

giordano commented:
CC @gbaraldi in case this helps with https://github.com/JuliaLang/julia/issues/48020

gbaraldi commented:
I tried just building from the LLVM repo, but I guess I'm missing some patches, so when yggdrasil is back soon :tm: I will try this out.

giordano commented:
Building this in codespaces I get
```
[15:24:52] CMake Error at /workspace/srcdir/cmake/Modules/HandleOutOfTreeLLVM.cmake:6 (message):
[15:24:52]   The provided LLVM_PATH (/workspace/srcdir/cmake/Modules/../../llvm) is not
[15:24:52]   a valid directory.  Note that building libc++ outside of the monorepo is
[15:24:52]   not supported anymore.  Please use a Standalone build against the monorepo,
[15:24:52]   a Runtimes build or a classic monorepo build.
[15:24:52] Call Stack (most recent call first):
[15:24:52]   CMakeLists.txt:46 (include)
[15:24:52] 
[15:24:52] 
[15:24:52] -- Configuring incomplete, errors occurred!
[15:24:52] See also "/workspace/srcdir/libunwind-14.0.6.src/build/CMakeFiles/CMakeOutput.log".
[15:24:52] See also "/workspace/srcdir/libunwind-14.0.6.src/build/CMakeFiles/CMakeError.log".
```

giordano commented:
For the record, compilation is successful, at least for aarch64-apple-darwin, if using as source the full llvm-project tarball, instead of the single libunwind one.

ararslan commented:
Does that mean we should actually try again to do https://github.com/JuliaPackaging/Yggdrasil/pull/3540? 😱 Or is it sufficient to switch from using the standalone source tarball for libunwind to the full LLVM one? For 12.0.1 I added a bundled patch that reverts the commit that disallowed standalone builds, but enough has changed since then that it'd be a real pain to adapt that patch. That's why for this one I just added a patch that deletes the error at the top and hoped for the best...

giordano commented:
> Or is it sufficient to switch from using the standalone source tarball for libunwind to the full LLVM one?

I downloaded the llvm-project tarball and it worked out smoothly. I did it inside the build environment, I didn't update the build script locally, but it shouldn't be _too_ hard

giordano commented on L/LLVMLibUnwind/LLVMLibUnwind@14.0.6/bundled/patches/0003-stdc-format-macros.patch:
```diff
@@ -0,0 +1,10 @@
+ #ifndef __DWARF_PARSER_HPP__
+ #define __DWARF_PARSER_HPP__
+ 
++#define __STDC_FORMAT_MACROS
```
They use it elsewhere, but not in libunwind: https://github.com/llvm/llvm-project/blob/5d87f60f894764bd435086e311605884638d2384/libcxx/include/inttypes.h#L244-L249

giordano commented:
License file is missing

ararslan commented:
> License file is missing

Should be fixed by https://github.com/JuliaPackaging/Yggdrasil/pull/6041/commits/3ad149b02ce2e6ae9a3256246a55c9c0634206ed#diff-3358a3924953c17aa8cedfd626863e654e065c5c0bc1f087fe462d13514c998aR63

ararslan commented:
@gbaraldi worth another try now I think 🙂
